### DISCUSSION TITLE
posts/mobile-homelab/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#267] [Comments]
# posts/mobile-homelab/

This has been months in the making, my new Mobile HomeLab! It’s a device that I can take with me to provide secure internet access for all of my devices. Not only can it provide secure access, but it can also let me bring apps and services with me when I travel. It’s built on Proxmox, OpenWRT, Pi-hole, and many other services. I’m taking this with me everywhere!

https://technotim.live/posts/mobile-homelab/

**Comment by vladislove41** [comment]
Hello. That's what I was looking for. I can't set up an access point. I have a simple mini pc with Intel Dual Band Wireless-AC 3165. Do I need to pass a network card to a virtual machine? (Hardware - Add - PCI device)?

**Reply by timothystewart6** [reply]
If you want to pass the WiFi controller to OpenWRT, yes, that's how you do it in Proxmox!

**Comment by brightplastik** [comment]
Hi Tim! 
First of all I want to thank you so much for the insight. Since I bought a second hand chinese firewall (no-brand intel 3865U 6x Gigabit RJ45) , I realized there's so much going on with the several possible implementations. Too much, I'd say!
Is there an article in which you guide through the setting of the internal NICs?
I think I really need some help here.
I'm stuck at the point in which you can log into openwrt, but of the 6 (maybe 5 remaining) ethernet ports there is no sign in the "Interfaces" tab. I only have eth0 and eth1, paired (I guess) with the net0 and net1 that were created following the guide by i12bretro you linked.
How do I setup the physical RJ45 ports to act like WAN and LAN ports of the Openwrt VM? I reckon I can play with 5 of the total 6, since vmbr0 with enp1s0 will be bonded to the proxmox UI, am I right? And eventually I should be able to remove the old interfaces and use the new ones withou loosing access to the VM, right? But how?

**Reply by timothystewart6** [reply]
I think that will work.  I don't have a lot of experience with OpenWRT outside of this and some routers that shipped with it, this was really my first time installing it myself and using it.  In my  experience it will detect all NICs and assign them and if it does not it means that it does not have the drivers to do so!

**Comment by Tjsyl** [comment]
Hey Tim!

I got my 2420 today and ordered bare bones. SP 16GB ddr4, 512gb m.2 and a 1tb sata PNY I had laying around, amazon had em on sale a while back.  I ordered the AX200 you had in your list off amazon but am having a hell of a time getting it to show up in OpenWRT. (shows up on the pci bus 06:10:0, 0000:05.0.0 in proxmox) How did you get it working?  I found this https://forum.openwrt.org/t/set-up-ax210-step-by-step/134415 but only end up with "Generic unknown" devices under the wireless and search finds nothing.  Following https://kb.protectli.com/kb/how-to-configure-protectli-wifi-kit-wap01k-in-openwrt-on-the-vault/ hasn't gotten it anywhere either, although I'm sure that firmware is only for their wifi kit.  OpenWRT, Software has a AX200 .ipk but that was the first thing I tried.  If anyone else has had luck please lend a hand.  May just have to order their wifi AC kit.

**Comment by pedrotchang** [comment]
I am having such a hard time! For some reason, after I follow the guide you linked, it works at first, except PCI passthrough doesn't seem to work.

Also, if I reboot, OpenWRT loses internet access. I tried adding my gateway, and dns, but still does not work. I followed this video: https://www.youtube.com/watch?v=Rl0ZS9DvHKA

**Comment by DMD770** [comment]
Hello!

I was wondering the homelab router you used, Protectli VP2420. did it had more than 1 slots to install like Sim card for Cellular service + wifi? because i have ordered this n100 from topton, https://www.aliexpress.us/item/3256806044774172.html. it stated the option is choose one, 3g,lte, or wifi. and it expansion is only "1xMPCIE wireless slot, support WiFi/4G/3G(3 choose 1) connection." so in this case it meant that i cannot do Wifi + LTE at the same time, i would have to use LTE then use both of USB port for Wifi AP and client option or remove the nvme, convert nvme to another MPCIE slot? i could go Nvme route but i already have that populated.

let me know if my thinking is correct.

Thanks!
